common/src/main/java/org/figuramc/figura/lua/api/PartCollection.java

```diff
@@ -22,12 +22,42 @@
 public class PartCollection implements MutablePart<PartCollection> {
     private Supplier<LuaTypeManager> manager;
     public final Set<FiguraModelPart> parts;
+    public final FiguraModelPart parent;
+    public String name;
```

Comment:
collections should be immutable
```suggestion
    public final String name;
```